guides/CONTRIBUTING.md

```diff
@@ -18,26 +16,158 @@ When filing an issue, make sure to provide:
+
+> [!CAUTION]
+> Currently, in this setup we have some code duplications, we planing soon to deprecate native-stack v5 and remove it from react-native-screens
+
+#### Changes in native code 
+
+If your change is only related to native code, for example you're fixing a bug, you **do not need** to post PR to `react-navigation`. `react-native-screens` is separate library installed alongside `react-navigation`, you can just bump `react-native-screens` version in you package.json file.
+
+#### Changes in JS code or both
+
+If you're changing native-stack v5 (`./native-stack`) you **do not need** to post PR to `react-navigation`.
+
+If you're changing native-stack v6, v7 you **need to** post PR to `react-navigation`, as the code belongs there now.
 
-### Pull requests
+If you're changing core functionality:
+- If it's a bug fix and does not change the interface then you **do not need** to post PR to `react-navigation`.
+- If you're adding new feature or changing API you **need to** post PR to `react-navigation`.
 
-Pull requests are more than welcome 🚀
+> [!TIP]
+> As the rule of thumb, if you're changing the external interface you need to prepare PRs for react-navigation
 
-We've created a template to help you with the process of submitting a pull request to this project. Remember to change or add documentation as needed and provide a code snippet (preferably added to `TestsExample/` project) that the developers could test your change on. In the PR template we've added a checklist for all the places where the documentation should be updated.
+### What is the flow to integrate with react-navigation
 
-If it's your first pull request [this guide](https://github.com/MarcDiethelm/contributing/blob/master/README.md) might be helpful for you.
+When you propose the changes that require posting PR to react-navigation. Please follow those steps:
 
-### Naming convensions
+1. Create PR with `react-native-screens` changes
+2. Create PR that changes the native-stack in `react-navigation`
+3. Wait for both to pass the review and have all all checks passed
+4. Merge `react-native-screens` changes
+5. Merge `react-native-navigation` changes
+6. Bump the version of main branch index ref in `react-native-screens` 
+7. Post and merge ASAP the main branch ref bump of `react-navigation`
 
-For commits and pull request names we follow a [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
+> [!WARNING]
+> Those steps are crucial, if you change the API in react-native-screens and won't merge react-navigation changes the libraries may go out of sync and crash i.e. because of not existing property. On the other hand, if you don't perform step 6 and 7, test examples and showcase app may stop working in react-native-screens.
```

Comment:
```suggestion
> Those steps are crucial, since changing the API in react-native-screens and won't merging react-navigation changes may cause in going both libraries out of sync and crashing (i.e. because of not existing property) //I don't quite like that sentence. On the other hand, if you don't perform step 6 and 7, test examples and showcase app may stop working in react-native-screens.
```

Reply:
Fixed in https://github.com/software-mansion/react-native-screens/pull/2140/commits/4e7c2cf577b090f42d690add4414ea22360f106a